contacts.install

```diff
@@ -17,18 +17,28 @@ function contacts_requirements($phase) {
     $themeHandler = \Drupal::service('theme_handler');
 
     if (!$themeHandler->themeExists('contacts_theme')) {
-      // Try to install theme.
-      $success = \Drupal::service('theme_installer')->install(['contacts_theme']);
-
-      if (!$success) {
-        $requirements['theme'] = array(
-          'title' => t('Missing Theme'),
-          'value' => 'To enable the contacts module you must first enable the contacts theme.',
-          'severity' => REQUIREMENT_ERROR,
-        );
-      }
+      $requirements['contacts_theme_missing'] = array(
+        'title' => t('Contacts theme missing'),
+        'value' => 'To enable the contacts module you must first enable the contacts theme.',
+        'severity' => REQUIREMENT_ERROR,
+      );
     }
   }
 
   return $requirements;
 }
+
+/**
+ * Implements hook_module_preinstall().
+ *
+ * Module config depends on a layout provided by the theme.
+ */
+function contacts_module_preinstall() {
```

Comment:
[`hook_module_preinstall($module)`](https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Extension%21module.api.php/function/hook_module_preinstall/8.2.x) is called on all installed modules whenever a module is installed, so we need to:

1) Check that this get's called (I'm pretty sure it is as it's added to the module list before the hook is invoked).
2) Add the argument and check that is is `contacts` being installed.
3) Does our test provide coverage for this process?